### PR TITLE
RHINENG-7661: Document "edge" hosts hidden by default

### PR DESCRIPTION
**Pull request by thearifismail**
# Overview

This PR is being created to address [RHINENG-7661](https://issues.redhat.com/browse/RHINENG-7661).

Mentioning "feature flag" in api call did not look right, so I did not included it.  If we think that it should be included, then the description in swagger/api doc should be:
`Read the entire list of all hosts available to the account, except "edge" hosts, which are hidden by default.  To get "edge" hosts, either use "?filter[system_profile][host_type][eq]=edge" or set the feature flag, "hbi.api.hide-edge-by-default", to "false"`

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ paths:
         - hosts
       summary: Read the entire list of hosts
       description: >-
-        Read the entire list of all hosts available to the account.
+        Read the entire list of all hosts available to the account, except "edge" hosts, which are hidden by default.  To get "edge" hosts, use "?filter[system_profile][host_type][eq]=edge"
```
Comment (kruai) [inline swagger/api.spec.yaml:15]: I think this information makes more sense in `/components/parameters/filter_param`. It applies to the system profile filter specifically, which isn't exclusive to the GET /hosts endpoint.

**Review by kruai — CHANGES_REQUESTED** [review]

**Comment by kruai** [comment]
> Mentioning "feature flag" in api call did not look right, so I did not included it. If we think that it should be included, (...)

Nope - we def don't want to encourage API users to flip our feature flags. It would be confusing to users since most of them don't know about, or have access to, our feature flags.

File: swagger/api.spec.yaml (hunk 1 above)
Comment (thearifismail) [reply swagger/api.spec.yaml:15]: @kruai please look at this again.  Thanks.

File: swagger/api.spec.yaml [hunk 2]
```diff
@@ -1245,10 +1245,17 @@ components:
...
         <br /><br />
         which equates to the URL param:
         <br /><br />
+        &nbsp;&nbsp;&nbsp;&nbsp;"?filter[system_profile][sap_system][eq]=true"
+        <br /><br />
+        The hosts provided by this filter do not include hosts of type "edge".
+        <br /><br />
+        To get "edge" hosts, explicit filter, like:
+        <br /><br />
+        &nbsp;&nbsp;&nbsp;&nbsp;{"system_profile": {"host_type": {"eq": "edge"}}} must be provided.  It equates to the URL param:
```
Comment (kruai) [inline swagger/api.spec.yaml:1258]:
```suggestion
        Hosts with host_type "edge" are filtered out by default.
        <br /><br />
        To get "edge" hosts, use this explicit filter:
        <br /><br />
        &nbsp;&nbsp;&nbsp;&nbsp;{"system_profile": {"host_type": {"eq": "edge"}}}
        <br /><br />
        which equates to the URL param:
        <br /><br />
```

File: swagger/api.spec.yaml (hunk 1 above)
Comment (kruai) [reply swagger/api.spec.yaml:15]: OK, I left a suggestion for that block. The change to this endpoint's description should still be reverted though

**Review by kruai — CHANGES_REQUESTED** [review]

**Review by kruai — APPROVED** [review]

**Comment by fstavela** [comment]
I reviewed the failing test pipelines and they are failing because of a failing pre-commit check:
```
15:51:16 Reorder python imports...................................................Passed
15:51:17 pyupgrade................................................................Passed
15:51:20 black....................................................................Passed
15:51:20 trim trailing whitespace.................................................Passed
15:51:20 fix end of files.........................................................Passed
15:51:20 check yaml...............................................................Passed
15:51:20 debug statements (python)................................................Passed
15:51:22 flake8...................................................................Passed
15:51:22 swagger-validation.......................................................Failed
15:51:22 - hook id: swagger-validation
15:51:22 - files were modified by this hook
15:51:22 
15:51:22 Created swagger/openapi.json from swagger/api.spec.yaml
15:51:22 
15:51:22 swagger-validation.......................................................Passed
15:51:32 =====================================
15:51:32 ====  ��� ERROR: PRECOMMIT FAILED  ====
15:51:32 =====================================
```

**Comment by thearifismail** [comment]
@fstavela this could be related to [PR 1637](https://github.com/RedHatInsights/insights-host-inventory/pull/1637)

**Review by kruai — APPROVED** [review]

**Pull request merged**